### DISCUSSION TITLE
Index Maintenance -  Rebuild only clustered indexes

### DISCUSSION THREAD

**Discussion by F-Flintstone** [olahallengren/sql-server-maintenance-solution#587] [Questions & Answers]
I'm looking for a simple method using Index Optimize procedure to rebuild only the clustered index on every table.  I'm also looking for the opposite.  How to rebuild all non-clustered indexes. 

My issue, there is no pattern that I can use to exclude or include.  Does anyone have a suggestion?

**Comment by BHubartt38** [comment]
Use [sys].[indexes] and [sys].[tables] to build a list of indexes that you can put into the @Indexes parameter.  You are going to need to do this prior to executing the index optimize procedure.  From [sys].[tables] you are going to want to use [type_desc] = 'USER_TABLE' and from [sys].[indexes] [type_desc] = 'CLUSTERED' or 'NONCLUSTERED'

Use the @Databases and @Indexes as shown in example (H) ( https://ola.hallengren.com/sql-server-index-and-statistics-maintenance.html )

**Reply by F-Flintstone** [reply]
I understand that doing query on sys.indexes will provide a list.  I'm trying to avoid doing that.  I'm looking for a switch to say ignore clustered indexes.

**Reply by BHubartt38** [reply]
Understood.  I don't think that there is anything currently available.  You could create an issue and/or make a request for a change.

Otherwise, the only way to get what you want is to work with what we currently have...

**Reply by ndevdariani** [reply]
It would be nice if this switch (for example @IndexType=1,2) could be added to the script.
